internal/controller/provider_controller.go

```diff
@@ -254,8 +255,12 @@ func (r *ProviderReconciler) validateCredentials(ctx context.Context, provider *
 	}
 
 	factory := notifier.NewFactory(address, proxy, username, provider.Spec.Channel, token, headers, certPool, password, string(provider.UID))
-	if _, err := factory.Notifier(provider.Spec.Type); err != nil {
+	if n, err := factory.Notifier(ctx, provider.Spec.Type); err != nil {
 		return fmt.Errorf("failed to initialize provider, error: %w", err)
+	} else if closer, ok := n.(io.Closer); ok {
```

Comment:
In this case it's more readable to do
```suggestion
	n, err := factory.Notifier(ctx, provider.Spec.Type)
	if err != nil {
		return fmt.Errorf("failed to initialize provider, error: %w", err)
	}
	if closer, ok := n.(io.Closer); ok {
```

Reply:
fixed, thanks!